google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/model.scala

```diff
@@ -77,66 +77,168 @@ object PubSubConfig {
     apply(projectId, clientEmail, privateKey, pullReturnImmediately, pullMaxMessagesPerInternalBatch)(actorSystem)
 }
 
+final class PublishMessage private (val data: String, val attributes: Option[immutable.Map[String, String]]) {
+  override def toString: String = "PublishMessage(data=" + data + ",attributes=" + attributes.toString + ")"
+
+  override def equals(other: Any): Boolean = other match {
+    case that: PublishMessage => data == that.data && attributes == that.attributes
+    case _ => false
+  }
+
+  override def hashCode: Int = java.util.Objects.hash(data, attributes)
+}
+
+object PublishMessage {
+  def apply(data: String, attributes: immutable.Map[String, String]) = new PublishMessage(data, Some(attributes))
+  def apply(data: String, attributes: Option[immutable.Map[String, String]]) = new PublishMessage(data, attributes)
+  def apply(data: String) = new PublishMessage(data, None)
+  def create(data: String) = new PublishMessage(data, None)
+  def create(data: String, attributes: immutable.Map[String, String]) = new PublishMessage(data, Some(attributes))
```

Comment:
Add a following scaladoc for the Java API methods:

```
/**
 * Java API
 */
```

Also Java API methods should take Java classes. Replace `immutable.Map` with `java.util.Map`. Then convert Java map to Scala immutable map with `attributes.asScala.toMap`.